File: front/src/models/dtos/experience/createExperienceDto.ts
import { Address } from '@/models/entities/address';
import { Company } from '@/models/entities/company';

/**
 * DTO for creating a new experience.
 */
export type CreateExperienceDto = {
  /**
   * The start date of the experience.
   * Formatted as an ISO 8601 string.
   */
  startDate: string;

  /**
   * The end date of the experience.
   * Formatted as an ISO 8601 string.
   */
  endDate: string;

  /**
   * The address of the experience.
   */
  address: Address;

  /**
   * The company where the experience was made.
   */
  company: Omit<Company, 'id'>;

  /**
   * The job id of the experience.
   */
  jobId: string;
};

File: backend/api_gateway/src/main/kotlin/com/linkedout/backend/dto/experience/CreateExperienceDto.kt
package com.linkedout.backend.dto.experience

import com.linkedout.backend.model.Address
import java.time.LocalDate

data class CreateExperienceDto(
    val startDate: LocalDate,
    val endDate: LocalDate,
    val address: Address,
    val company: Company,
    val jobId: String
) {
    data class Company(
        val name: String
    )
}
